### ISSUE TITLE
Unable to Open Virtual Keyboard in 5.6.6

### ISSUE THREAD

**Issue by BlueyDragon** [BlitterStudio/amiberry#1239]
I'm using Amiberry v5.6.6 in a Retroarch environment - an arcade machine running with Retropie on a Raspberry Pi 3, Raspbian version buster. The machine has two arcade control encoders that are recognized as "DragonRise Generic USB Joystick" in the controls menu, and the [R] is showing that they're being mapped by Retroarch.

I haven't found any way to display the onscreen keyboard to use it. The "Virtual Keyboard enabled" option is checked in the Virtual Keyboard and the hotkey is set to Select, but pressing Select does nothing. I've also attempted to check the option saying "Use Retroarch Vkbd button" but that does not work either. I assume the Vkbd button it's talking about is "Show Keyboard Overlay" in Retroarch's configuration, which I do have set to the hotkey enable button and a "button 9" which is supposed to be one of the buttons on the front of the machine.

The joysticks do work - the Retroarch hotkeys for opening the options (which opens the Amiberry GUI) and for quitting are both working, and I'm able to use the joysticks and primary fire button in games. It seems that the only thing that isn't registering at all is the virtual keyboard.

**Comment by giantclambake** [comment]
Can't speak specifically to your setup (don't have Dragonrise devices nor retroarch environment), but I did check this in v5.6.8 on x86-64 and it appears to work as expected 'natively' when using amiberry standalone, mapping a xpad button to custom control 'On-screen keyboard' with Virtual kbd enabled ; emulation starts with vkbd displayed, mapped button toggles it on/off.

Looking at the amiberry & retroarch sources, in theory the following line should appear in your retroarch.cfg (using the button# you chose)...

input_osk_toggle_btn = "9"

You should check if that's the case - amiberry is trying to parse the value of `input_osk_toggle_btn`

**Comment by giantclambake** [comment]
I just noticed that in #1249 the user there also reports that Virtual Keyboard wasn't working for them in v5.6.6 with a similar setup, but this started working for them after upgrading to v5.6.8 ~ perhaps you should try upgrading amiberry to v5.6.8 and see if you it works for you as well?

**Comment by BlueyDragon** [comment]
I can confirm that it still happens in 5.6.6 even when retroarch.cfg has the `input_osk_toggle_btn = "9"` line in it. Also, the default selection of `select` doesn't seem to work either.

Upgrading to 5.6.8 looks to be the best bet, but so far Retropie still has 5.6.6 listed as the "latest". I tried changing the version number manually in `amiberry.sh` in the scriptmodules folder, which seems to be what actually controls what version number of a given emulator you get, but then an additional patch failed... I think it was preserve_env or something to that effect? I did download the latest release for a Raspbian Buster RP3 environment, but is there a particular path I should copy all the files to?
